Helix crashes when launched from `zk` application

### Summary

When [`zk`](https://github.com/mickael-menu/zk) tries to run helix, helix crashes with error status 101. 

I wanted to add helix as default editor for [`zk`](https://github.com/mickael-menu/zk). For fresh install I set editor in [`zk` config](https://github.com/mickael-menu/zk/blob/main/docs/tool-editor.md) but 

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. Fresh install of  [`zk`](https://github.com/mickael-menu/zk)
2. Set `editor = "hx"` in config per [config documentation](https://github.com/mickael-menu/zk/blob/main/docs/tool-editor.md)
3. Invoke any action in `zk` that should cause launch of helix, for example `zk new` 

I expected this to happen:
Helix to launch

Instead, this happened:
Helix crashes with:
`zk: error: failed to launch editor: hx /Users/<snip>/5e19.md: exit status 101`

Here is the code of `zk` that [launches editor](https://github.com/mickael-menu/zk/blob/50855154e2f2878994e7a99e2ec2619addaeee90/internal/adapter/editor/editor.go#L41). It works for me with `nano` and `vim`

Log attached that corresponds with launching running `zk new`.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-04-14T21:22:22.528 helix_view::editor [ERROR] Failed to initialize the LSP for `source.md` { LSP not defined }
```

</details>


### Platform

macOS

### Terminal Emulator

iTerm

### Helix Version

helix 22.03 (d4e45fd4)

--- a/helix-term/Cargo.toml
+++ b/helix-term/Cargo.toml
@@ -37,7 +37,7 @@ which = "4.4"
 
 tokio = { version = "1", features = ["rt", "rt-multi-thread", "io-util", "io-std", "time", "process", "macros", "fs", "parking_lot"] }
 tui = { path = "../helix-tui", package = "helix-tui", default-features = false, features = ["crossterm"] }
-crossterm = { version = "0.26", features = ["event-stream", "use-dev-tty"] }
+crossterm = { version = "0.26", features = ["event-stream"] }
 signal-hook = "0.3"
 tokio-stream = "0.1"
 futures-util = { version = "0.3", features = ["std", "async-await"], default-features = false }
@@ -69,6 +69,9 @@ grep-searcher = "0.1.11"
 [target.'cfg(not(windows))'.dependencies]  # https://github.com/vorner/signal-hook/issues/100
 signal-hook-tokio = { version = "0.3", features = ["futures-v0_3"] }
 libc = "0.2.146"
+
+[target.'cfg(target_os = "macos")'.dependencies]
+crossterm = { version = "0.26", features = ["event-stream", "use-dev-tty"] }
 
 [build-dependencies]
 helix-loader = { version = "0.6", path = "../helix-loader" }

--- a/helix-term/Cargo.toml
+++ b/helix-term/Cargo.toml
@@ -37,7 +37,7 @@ which = "4.4"
 
 tokio = { version = "1", features = ["rt", "rt-multi-thread", "io-util", "io-std", "time", "process", "macros", "fs", "parking_lot"] }
 tui = { path = "../helix-tui", package = "helix-tui", default-features = false, features = ["crossterm"] }
-crossterm = { version = "0.26", features = ["event-stream"] }
+crossterm = { version = "0.26", features = ["event-stream", "use-dev-tty"] }
 signal-hook = "0.3"
 tokio-stream = "0.1"
 futures-util = { version = "0.3", features = ["std", "async-await"], default-features = false }

--- a/helix-term/src/application.rs
+++ b/helix-term/src/application.rs
@@ -215,11 +215,6 @@ impl Application {
             }
         } else if stdin().is_tty() || cfg!(feature = "integration") {
             editor.new_file(Action::VerticalSplit);
-        } else if cfg!(target_os = "macos") {
-            // On Linux and Windows, we allow the output of a command to be piped into the new buffer.
-            // This doesn't currently work on macOS because of the following issue:
-            //   https://github.com/crossterm-rs/crossterm/issues/500
-            anyhow::bail!("Piping into helix-term is currently not supported on macOS");
         } else {
             editor
                 .new_file_from_stdin(Action::VerticalSplit)

